### ISSUE TITLE
Create seedlot number api on backend and oracle-api

### ISSUE THREAD

**Issue by RMCampos** [bcgov/nr-spar#237]
Create API for registering a new seedlot number, having it saved both on `backend` and `oracle-api`.

**Comment by RMCampos** [comment]
POST - /api/seedlots
```json
{
  "applicantClientNumber": "MOF",
  "applicantLocationCode": "555",
  "applicantEmailAddress": "ricardo.decampos@gov.bc.ca",
  "vegetationCode": "FDI",
  "seedlotSourceCode": "TPT",
  "intendedForCrownLand": true,
  "sourceInBc": true
}
```

It will return a json:
```json
{
  "nextSeedlotNumber": 123456
}
```

**Comment by RMCampos** [comment]
Notes on the next seedlot number (from current seedlot SP):

```sql
SELECT MAX(TO_NUMBER(s.SEEDLOT_NUMBER))
FROM THE.SEEDLOT s
WHERE s.SEEDLOT_NUMBER
	BETWEEN 53000 AND 59998; -- 53966

-- CONST_CLASS_B_LOTNUM_MIN=CONSTANT VARCHAR2(5) := '53000';
-- CONST_CLASS_B_LOTNUM_MAX=CONSTANT VARCHAR2(5) := '59999';
-- CONST_CLASS_A_LOTNUM_MIN      CONSTANT VARCHAR2(5) := '63000';
-- CONST_CLASS_A_LOTNUM_MAX      CONSTANT VARCHAR2(5) := '69999';

```

**Comment by thiagosanchezenc** [comment]
Comments on database package to be used on this task:

In order to create and lock a new seedlot number there are a few steps that needs to be executed:
- First call the init procedure to set all attributes to null;
- Then set all relevant attributes by calling the set procedures for each attribute;
- Finally call the add procedure that generates the seedlot number and inserts the record on SEEDLOT table.

After adding a new seedlot by providing only the Species, Genetic Class, and Applicant Agency (default values), those are the columns populated on SEEDLOT table:
- SEEDLOT_NUMBER
- SEEDLOT_STATUS_CODE
- VEGETATION_CODE
- GENETIC_CLASS_CODE
- REGISTERED_SEED_IND = N
- UTM_ZONE = 0
- UTM_EASTING = 0
- UTM_NORTHING = 0
- ORIGINAL_SEED_QTY = 0
- STORED_CLI_NUMBER
- STORED_CLI_LOCN_CD
- APPLICANT_CLIENT_NUMBER
- APPLICANT_CLIENT_LOCN
- ENTRY_USERID
- ENTRY_TIMESTAMP
- UPDATE_USERID
- UPDATE_TIMESTAMP
- REVISION_COUNT = 2

**Comment by RMCampos** [comment]
Moving this task to blocked. Due to still be waiting for the package access. Follow the conversation here: https://teams.microsoft.com/l/message/19:deff5cb6c46f45e3b84c1b786b1799c1@thread.tacv2/1689359595371?tenantId=6fdb5200-3d0d-4a8a-b036-d3685e359adc&groupId=16e64d26-4d6f-422e-8bf1-7e7c5a33c7de&parentMessageId=1689359595371&teamName=Forestry%20Digital%20Services&channelName=Database&createdTime=1689359595371&allowXTenantAccess=false

**Comment by RMCampos** [comment]
Right. It was decided to create the seedlot number on Postgres and Backend API only, for now, and later on the ETL workflows create on Oracle THE.

Note: We might start a new range to keep track of the new numbers, but that's not decided yet. 
Task unblocked now.
